### PR TITLE
feat(pm_list): add required fields for google pay

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Resolves [#3706](https://github.com/juspay/hyperswitch-cloud/issues/3706)

### Test Cases

Must be tested for each of this connectors

1. Adyen
```
None
```
2.Bluesnap
```
None
```
3.Noon
```
None
```
4.Nuvei
```
None
```
5.Airwallex
```
None
```
6.Authorizedotnet
```
None
```
7.Checkout
```
None
```
8.Globalpay
```
None
```
9.Multisafepay
```
Required fields: 
     billing.address.first_name
     billing.address.last_name
     billing.address.city
     billing.address.state
     billing.address.zip
     billing.address.country
     billing.address.line1
     billing.address.line2
```
10.PayU
```
None
```
11.Rapyd
```
None
```
12.Stripe
```
None
```
13.Trustpay
```
None
```

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by prasunna09 — COMMENTED** [review]
here you have only mentioned about connector list that supports google pay mandates. but what about the required fields? Are there no required fields for this pm type?

**Comment by AkshayaFoiger** [comment]
> here you have only mentioned about connector list that supports google pay mandates. but what about the required fields? Are there no required fields for this pm type?

Most of the connectors don't require additional fields, but some does. I will update the PR

File: crates/router/src/configs/defaults.rs [hunk 1]
```diff
@@ -4897,6 +4910,146 @@ impl Default for super::settings::RequiredFields {
...
+                                                field_type: enums::FieldType::UserAddressLine1,
+                                                value: None,
+                                            }
+                                        ),
+                                        (
+                                            "billing.address.line2".to_string(),
+                                            RequiredFieldInfo {
+                                                required_field: "billing.address.line2".to_string(),
+                                                display_name: "line2".to_string(),
+                                                field_type: enums::FieldType::UserAddressLine1,
```
Comment (prasunna09) [inline crates/router/src/configs/defaults.rs:5046]:
```suggestion
                                                field_type: enums::FieldType::UserAddressLine2,
```

**Review by prasunna09 — CHANGES_REQUESTED** [review]

File: crates/router/src/configs/defaults.rs [hunk 2]
```diff
@@ -4803,7 +4816,7 @@ impl Default for super::settings::RequiredFields {
                         ConnectorFields {
                             fields: HashMap::from([
                                 (
-                                    enums::Connector::Stripe,
+                                    enums::Connector::Adyen,
                                     RequiredFieldFinal {
                                         mandate: HashMap::new(),
```
[the remark this answers is not in the document]
Comment (prasunna09) [reply crates/router/src/configs/defaults.rs:4821]: My bad, email is required for Paypal mandates for Adyen, not for googlepay.

**Review by prasunna09 — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by Chethan-rao — APPROVED** [review]

**Pull request merged**